lib/httpserver/httpserver.go

```diff
@@ -429,6 +433,11 @@ var (
 	pprofDefaultRequests = metrics.NewCounter(`vm_http_requests_total{path="/debug/pprof/default"}`)
 	faviconRequests      = metrics.NewCounter(`vm_http_requests_total{path="/favicon.ico"}`)
 
+	successfulAuthAtemptsTotal      = metrics.NewCounter(`vm_auth_requests_total{attempt="success", type="basic"}`)
+	unSuccessfulAuthAtemptsTotal    = metrics.NewCounter(`vm_auth_requests_total{attempt="failure", type="basic"}`)
+	successfulAuthKeyAtemptsTotal   = metrics.NewCounter(`vm_auth_requests_total{attempt="success", type="authkey"}`)
+	unSuccessfulAuthKeyAtemptsTotal = metrics.NewCounter(`vm_auth_requests_total{attempt="failure", type="authkey"}`)
+
```

Comment:
What is the expected usage for this metrics? Could you provide an example of alerting rule which would notify user about something suspicious?

Reply:
These metrics would help maintainer know how many successful/unsuccessful authentication attempts. If unsuccessful attempts are more, then it might need to be checked why many unsuccessful attempts are being made and some corrective action can be taken.

Reply:
I think it would be more straightforward to provide the following metrics then:
```
	authBasicRequestErrors = metrics.NewCounter(`vm_http_request_errors_total{path="*", reason="wrong basic auth creds"}`)
	authKeyRequestErrors = metrics.NewCounter(`vm_http_request_errors_total{path="*", reason="wrong auth key"}`)
```
In this way we keep metric names and labels consistent with other metrics.

> These metrics would help maintainer know how many successful/unsuccessful authentication attempts.

With these metrics you can than compare `increase(vm_http_requests_total)` with `increase(vm_http_request_errors_total{path!="unsupported"})` to get the % of requests that failed authentification.

Reply:
Hi @hagen1778, i will modify to align to above.